src/vs/editor/common/services/modeService.ts

```diff
@@ -20,6 +20,7 @@ export interface ILanguageExtensionPoint {
 	aliases?: string[];
 	mimetypes?: string[];
 	configuration?: URI;
+	icon?: string;
```

Comment:
Sure, can do. I only ran into a problem when I changed the icon field on the ILanguageExtensionPoint from string to ThemeIcon because of monaco (monacod.d.ts) not seemingly containing/finding this class.